tests/test_operator.py

```diff
@@ -317,6 +317,26 @@ def test_nested_lowering(self):
         assert np.all(u0.data[:2, :2] == 1) and np.all(u0.data[1:3, 1:3] == 1)
         assert np.all(u0.data[2:3, 3] == 2) and np.all(u0.data[3, 2:3] == 2)
 
+    def test_nested_lowering_indexify(self):
+        """
+        Tests that nested function are lowered if only used as index.
+        """
+        grid = Grid(shape=(4, 4), dtype=np.int32)
+        x, y = grid.dimensions
+
+        u0 = Function(name="u0", grid=grid)
+        u1 = Function(name="u1", grid=grid, dtype=np.int32)
+
+        u0.data[:, :] = 2
+        u1.data[:, :] = 1
+
+        eq0 = Eq(u0._subs(x, u1), 2*u0)
```

Comment:
why not `subs` instead of `_subs`?

Reply:
Oh it does the same just quicker through our `_subs` that bypass some sympy check and stuff, just habit can switch it.